sdk/eventhub/azure-eventhubs/azure/eventhub/aio/consumer_async.py

```diff
@@ -146,6 +147,32 @@ def _create_handler(self):
             self.source = self.redirected.address
         await super(EventHubConsumer, self)._open(timeout_time)
 
+    async def _receive(self, timeout_time=None, max_batch_size=None, **kwargs):
+        last_exception = kwargs.get("last_exception")
+        data_batch = kwargs.get("data_batch")
+
+        await self._open(timeout_time)
+        remaining_time = timeout_time - time.time()
+        if remaining_time <= 0.0:
+            if last_exception:
+                log.info("%r receive operation timed out. (%r)", self.name, last_exception)
+                raise last_exception
```

Comment:
Is this called from with an exception handler? If not, we've lost context when raising the exception...

Reply:
The retry decorator function will catch the exception and keep record of the last exception